### PR TITLE
fix: replace selection polyfill in quill to handle Safari selection issues

### PR DESCRIPTION
**Pull request by sissbruecker**
## Description

Updates Quill dependency to use a new polyfill for shadow DOM text selection (related PR: https://github.com/vaadin/quill/pull/2).

Fixes: https://github.com/vaadin/web-components/issues/2115
Fixes: https://github.com/vaadin/web-components/issues/2139 - this one is about text input and selection not working in any Vaadin component in iOS 12, which is also caused by the old polyfill

### CODE REVIEW

File: test/index.html [hunk 1]
```diff
@@ -11,12 +11,7 @@
 
 <body>
   <script>
-    WCT.loadSuites(
-      window.RichTextEditorSuites
-        .reduce(function(suites, suite) {
-          return suites.concat([suite, `${suite}?wc-shadydom=true`]);
-        }, [])
-    );
+    WCT.loadSuites(window.RichTextEditorSuites);
```
Comment (sissbruecker) [inline test/index.html:14]: Running test suites with ShadyDOM enabled caused issues in Chrome.

Discussed with Serhii and decided that this was not necessary anymore. Browsers without shadow DOM support enable ShadyDOM by default. For other browsers we don't need ShadyDOM enabled, so this is not a valid use-case.

File: test/basic.html [hunk 2]
```diff
@@ -73,9 +73,7 @@
           });
 
           ['bold', 'italic', 'underline', 'strike', 'blockquote', 'code-block'].forEach(fmt => {
-            // Applying formatting to selected text is currently broken in Safari:
-            // https://github.com/vaadin/web-components/issues/2115
-            !isDesktopSafari && it(`should apply ${fmt} formatting to the selected text on click`, () => {
+            it(`should apply ${fmt} formatting to the selected text on click`, () => {
```
Comment (sissbruecker) [inline test/basic.html:76]: This was fixed by updating the shadow selection polyfill in the quill library.

**Review by hdamr — APPROVED** [review]
I checked RTE demo pages on various versions of iOS (10.3, 11.2, 12.1,13.3, 14.3) and it worked as expected and no problem was seen. Also other browsers (IE11, FF, Edge) were tested and it was working fine, both RTE functionality and also adding an extra input field on the page to test typing into input field.

**Review by web-padawan — APPROVED** [review]
LGTM 🎉

**Pull request merged**